static/src/javascripts-legacy/projects/common/modules/atoms/youtube.js

```diff
@@ -71,7 +75,41 @@ define([
         }
     }
 
+
+
+    function shouldAutoplay(){
+
+        function isMobile() {
+            return detect.isIOS() || detect.isAndroid();
+        }
+
+        function isInternalReferrer() {
+            /**
+             * Polyfill for String.startsWith
+             * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith
+             */
+            if (!String.prototype.startsWith) {
```

Comment:
this isn't really a good idea, I'm afraid... 

firstly, how would anyone else have any idea that it's been polyfilled in this module? if they wanted to polyfill it to, we'd be doing twice (at least).

secondly, you're changing the prototype of `String` directly, meaning that at any point before this code is evaluated, the _global_ `String` does not contain the `startsWith` method, and then after, it does. This introduces side affects to the entire environment, just so this string can be checked.

it's generally much better practise to create a method that returns the native method if exists.

however, for something that really only needs to be done once, it's not clear it even really needs polyfilling, especially when you could use the polyfill you've added directly, or maybe

```js
if(config.page.isDev) {
    return document.referrer.indexOf(window.location.origin) === 0;
}
else {
    return document.referrer.indexOf(config.page.host) === 0;
}
```